src/spi.rs

```diff
@@ -82,6 +93,45 @@ where
         Ok(FirmwareVersion::new(bytes)) // e.g. 1.7.4
     }
 
+    fn set_passphrase(&mut self, ssid: &str, passphrase: &str) -> Result<(), self::Error> {
+        self.control_pins.wait_for_esp_select();
+
+        self.send_cmd(NinaCommand::SetPassphrase, 2).ok().unwrap();
+
+        let ssid_param = NinaSmallArrayParam::new(ssid);
+        self.send_param(ssid_param);
+
+        let passphrase_param = NinaSmallArrayParam::new(passphrase);
+        self.send_param(passphrase_param);
+
+        self.send_end_cmd();
+
+        let command_size: u8 = 6 + ssid.len() as u8 + passphrase.len() as u8;
+        self.pad_to_multiple_of_4(command_size as u16);
+
+        self.control_pins.esp_deselect();
+        self.control_pins.wait_for_esp_select();
+
+        self.wait_response_cmd(NinaCommand::SetPassphrase, 1);
```

Comment:
Do you want to use `?` consistently here like you use on line 129?

Reply:
Same as above. I'd like to wait until https://github.com/Jim-Hodapp-Coaching/esp32-wroom-rp/issues/21 to think through this stuff in a more wholistic way.

Reply:
I'm cool with that.